packages/core/src/components/AdaptiveGrid/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import withStyles, { WithStylesProps } from '../../composers/withStyles';
+
+type State = {
+  items: number;
+};
+
+export type AdaptiveGridProps = {
+  /** List of width/item pairs, describes how many items to display for windows of at least that width. */
+  breakpoints?: { [key: number]: number };
+  /** Items per row for screens smaller than the smallest breakpoint. */
+  defaultItemsPerRow?: number;
+  /** Removes padding between items. */
+  noGutter?: boolean;
+};
+
+type Props = AdaptiveGridProps & WithStylesProps;
+
+class AdaptiveGrid extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    breakpoints: {},
+    defaultItemsPerRow: 1,
+    noGutter: false,
+  };
+
+  state = {
+    items: this.getItems(),
+  };
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  private handleResize = () => {
+    this.setState({
+      items: this.getItems(),
+    });
+  };
+
+  // Find the largest breakpoint smaller than the current width
+  getItems(): number {
+    const { breakpoints, defaultItemsPerRow } = this.props;
+    const width = window.innerWidth;
+
+    const max = Object.keys(breakpoints!).reduce((acc, breakpoint) => {
+      const b = parseInt(breakpoint, 10);
+
+      return b < width && b > acc ? b : acc;
+    }, -1);
+
+    return max > -1 ? breakpoints![max] : defaultItemsPerRow!;
+  }
+
+  render() {
+    const { children, cx, noGutter, styles } = this.props;
+    const { items } = this.state;
+
+    const itemStyle = {
+      width: `${100 / items}%`,
+    };
+
+    const childElements =
+      children &&
+      React.Children.map(children, (child: React.ReactNode, idx: number) => (
```

Comment:
We should check that child is truthy, otherwise we'd be wrapping an empty value with a `div`, which is probably not what we want.

```
if (!child) return null;
```